### PR TITLE
[ISSUE #12872]Refactor ConfigCache and related pathways ,uniformly use UTF-8 encoding for md5 fields

### PR DESCRIPTION
**Pull request by Sunrisea**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Refactor ConfigCache and related pathways ,uniformly use UTF-8 encoding for md5 fields. Utilize the SPI mechanism in related pathways and methods to ensure extensibility for support of other types of encoding. Fix test.

## Brief changelog

Create the ConfigCacheFactory interface to handle the creation of ConfigCache instances.
Create the Md5PostProcessor interface to handle the md5 dump of ConfigCache.
Create the Md5Comparator interface.
Utilize the SPI (Service Provider Interface) mechanism in related pathways and methods to ensure extensibility for support of other types of encoding, and implement the default handling methods for UTF-8 encoded md5 values.

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [*] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [*] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [*] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [*] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [*] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/12227715560)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

File: config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactory.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+     * @return the config cache gray
+     */
+    public ConfigCacheGray createConfigCacheGray();
+    
+    /**
+     * Gets config cache factroy name.
+     *
+     * @return the config cache factory name
+     */
+    public String getConfigCacheFactoryName();
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/model/ConfigCacheFactory.java:45]: change to  getName

File: config/src/main/java/com/alibaba/nacos/config/server/utils/Md5Comparator.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * @author Sunrisea
+ */
+public interface Md5Comparator {
+    
+    /**
+     * Gets md 5 comparator name.
+     *
+     * @return the md 5 comparator name
+     */
+    public String getMd5ComparatorName();
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/utils/Md5Comparator.java:36]: change to getName

**Comment by wuyfee** [comment]
$\color{green}{SUCCESS}$
[DETAILS](https://github.com/alibaba/nacos/actions/runs/12228641690)
 ✅ - docker:   success 
 ✅ - deploy (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-java-test (standalone & cluster & standalone_auth):   success 
 ✅ - e2e-go-test (standalone & cluster):   success 
 ✅ - e2e-cpp-test (standalone & cluster):   success 
 ✅ - e2e-csharp-test (standalone & cluster):   success 
 ✅ - e2e-nodejs-test (standalone & cluster):   success 
 ✅ - e2e-python-test (standalone & cluster):   success 
 ✅ - clean (standalone & cluster & standalone_auth):   success

**Review by shiyiyue1102 — APPROVED** [review]

**Pull request merged**